one/src/daemon.rs

```diff
@@ -434,8 +436,19 @@ pub async fn run(opts: DaemonOpts) -> Result<()> {
     let peer_svc = Arc::new(PeerService::new(sqlite_pool.clone()));
     let interest_svc = Arc::new(InterestService::new(sqlite_pool.clone()));
     let event_validation = opts.event_validation.unwrap_or(true);
+    let mut ss = shutdown_signal.resubscribe();
     let event_svc = Arc::new(
```

Comment:
this whole PR could probably be redone as simply this (which I didn't think of for some reason)... piping it down could allow for a chance to exit gracefully and clean up, but it's more or less implemented like this currently just lower

```rust
    let mut ss = shutdown_signal.resubscribe();
    let event_svc = tokio::select! {
        svc = EventService::try_new(sqlite_pool.clone(), true, event_validation, rpc_providers) => {
            Arc::new(svc?)
        }
        _ = ss.recv() => {
            bail!("received shutdown signal before startup tasks finished")
        }
    };
    ```